crates/re_arrow_store/tests/correctness.rs

```diff
@@ -307,64 +310,40 @@ fn gc_correct() {
-    //     Timeline::new("frame_nr", TimeType::Sequence),
-    //     MsgId::name(),
-    // );
-
-    // let row_ids = row_id_chunks
-    //     .iter()
-    //     .flat_map(|chunk| arrow_array_deserialize_iterator::<Option<MsgId>>(&**chunk).unwrap())
-    //     .map(Option::unwrap) // MsgId is always present
-    //     .collect::<ahash::HashSet<_>>();
-    // assert!(!row_ids.is_empty());
-
-    // if let err @ Err(_) = store.sanity_check() {
-    //     store.sort_indices_if_needed();
-    //     eprintln!("{store}");
-    //     err.unwrap();
-    // }
-    // check_still_readable(&store);
-    // for row_id in &row_ids {
-    //     assert!(store.get_msg_metadata(row_id).is_some());
-    // }
-
-    // store.clear_msg_metadata(&row_ids);
-
-    // if let err @ Err(_) = store.sanity_check() {
-    //     store.sort_indices_if_needed();
-    //     eprintln!("{store}");
-    //     err.unwrap();
-    // }
-    // check_still_readable(&store);
-    // for row_id in &row_ids {
-    //     assert!(store.get_msg_metadata(row_id).is_none());
-    // }
-
-    // let row_id_chunks = store.gc(
-    //     GarbageCollectionTarget::DropAtLeastPercentage(1.0),
-    //     Timeline::new("frame_nr", TimeType::Sequence),
-    //     MsgId::name(),
-    // );
-
-    // let row_ids = row_id_chunks
-    //     .iter()
-    //     .flat_map(|chunk| arrow_array_deserialize_iterator::<Option<MsgId>>(&**chunk).unwrap())
-    //     .map(Option::unwrap) // MsgId is always present
-    //     .collect::<ahash::HashSet<_>>();
-    // assert!(row_ids.is_empty());
-
-    // if let err @ Err(_) = store.sanity_check() {
-    //     store.sort_indices_if_needed();
-    //     eprintln!("{store}");
-    //     err.unwrap();
-    // }
-    // check_still_readable(&store);
-
-    // assert_eq!(2, store.total_temporal_component_rows());
+    let stats = DataStoreStats::from_store(&store);
+
+    let (row_ids, stats_diff) = store.gc(GarbageCollectionTarget::DropAtLeastPercentage(1.0));
+    assert_eq!(row_ids.len() as u64, stats.total_rows);
+    assert_eq!(
+        stats.metadata_registry_rows,
+        stats_diff.metadata_registry_rows
+    );
+    assert_eq!(
+        stats.metadata_registry_size_bytes,
+        stats_diff.metadata_registry_size_bytes
+    );
+    assert_eq!(stats.total_temporal_rows, stats_diff.total_temporal_rows);
+
+    if let err @ Err(_) = store.sanity_check() {
+        store.sort_indices_if_needed();
+        eprintln!("{store}");
+        err.unwrap();
+    }
+    check_still_readable(&store);
+    for row_id in &row_ids {
+        assert!(store.get_msg_metadata(row_id).is_none());
+    }
+
+    let (row_ids, stats_diff) = store.gc(GarbageCollectionTarget::DropAtLeastPercentage(1.0));
+    assert!(row_ids.is_empty());
+    assert_eq!(DataStoreStats::default(), stats_diff);
+
+    if let err @ Err(_) = store.sanity_check() {
+        store.sort_indices_if_needed();
+        eprintln!("{store}");
+        err.unwrap();
+    }
```

Comment:
If `Display` would call `sort_indices_if_needed` we could simplify this code quite a bit, here and in many places

```suggestion
    store.sanity_check().unwrap_or_else(|err| panic!("{store}\n\n{err}"));
```